.github/actions/make/android/lib/action.yml

```diff
@@ -32,8 +32,14 @@ runs:
 
     - uses: ./.github/actions/make
       with:
-        key: lib-${{ inputs.task }}-${{ github.sha }}
+        key: lib-${{ inputs.task }}
         make-rule: ${{ inputs.task }}
+        dependencies: |
+          keystore
+          mls-provider
+          crypto-macros
+          crypto
+          crypto-ffi
```

Comment:
You also need `Cargo.lock` here. Otherwise upgrading a dependency without updating any of our actual code would result in a false cache hit.